### ISSUE TITLE
0054 - Spiral Matrix

### ISSUE THREAD

**Issue by vJechsmayr** [vJechsmayr/PythonAlgorithms#161]
## Description of the Problem
Given a matrix of _m x n_ elements (_m_ rows, _n_ columns), return all elements of the matrix in spiral order.

**Example 1:**
```
Input:
[
 [ 1, 2, 3 ],
 [ 4, 5, 6 ],
 [ 7, 8, 9 ]
]
Output: [1,2,3,6,9,8,7,4,5]
```

**Example 2:**
```
Input:
[
  [1, 2, 3, 4],
  [5, 6, 7, 8],
  [9,10,11,12]
]
Output: [1,2,3,4,8,12,11,10,9,5,6,7]
```

## Code
```
class Solution:
    def spiralOrder(self, matrix: List[List[int]]) -> List[int]:
```

## Link To The LeetCode Problem
[LeetCode](https://leetcode.com/problems/spiral-matrix/)

**Comment by sus5pect** [comment]
I am interested in solving this can u please assign me

**Comment by divyhshah** [comment]
hey can you please assign this to me i want to do this in python

**Comment by Darshini-be** [comment]
Interested, Please assign this to me. Thanks

**Comment by divyhshah** [comment]
@vJechsmayr i already done this and also made a pr can you please merge it
Thanks

**Comment by divyhshah** [comment]
closes #161
